src/common/ring_buffer.rs

```diff
@@ -0,0 +1,118 @@
+use managed::Managed;
+
+pub trait Resetable {
+    fn reset(&mut self);
+}
+
+/// A ring buffer.
+#[derive(Debug)]
+pub struct RingBuffer<'a, T: 'a> {
+    storage: Managed<'a, [T]>,
+    read_at: usize,
+    length: usize,
```

Comment:
I would really like it if you at least kept the formatting as-is when moving code around. My preferences notwithstanding this makes git's copy/move tracking mechanisms work better.